File: src/game/entity/items/UsableItem.java
package game.entity.items;

import game.entity.models.BomberEntity;

public abstract class UsableItem {
    protected BomberEntity owner;

    public abstract void use();

    public void setOwner(BomberEntity owner) {
        this.owner = owner;
    }

    public void give() {
        owner.give(this);
    }

    public void remove() {
        owner.removeItem();
    }

    public BomberEntity getOwner() {
        return owner;
    }
}

File: src/game/items/UsableItem.kt
package game.items

import game.Bomberman
import game.entity.EntityTypes
import game.entity.player.BomberEntity
import game.http.events.forward.UseItemHttpEventForwarder

abstract class UsableItem {
    lateinit var owner: BomberEntity
    abstract fun use()
    abstract fun combineItems(item: UsableItem)
    abstract val imagePath: String?
    abstract val count: Int
    abstract val type: ItemsTypes

    fun give() {
        Bomberman.getMatch().give(owner, this)
    }

    fun remove() {
        Bomberman.getMatch().removeItem(owner)
    }
}